rskj-core/src/test/java/org/ethereum/vm/trace/DetailedProgramTraceProcessorTest.java

```diff
@@ -71,19 +94,50 @@ public void setAndGetTraceAsJsonNode() {
         DetailedProgramTrace trace = buildTestTrace(ownerDW);
         processor.processProgramTrace(trace, hash);
 
-        JsonNode jnode = processor.getProgramTraceAsJsonNode(hash);
+        Keccak256 hash2 = TestUtils.randomHash();
+        DataWord ownerDW2 = DataWord.valueOf(42);
+        DetailedProgramTrace trace2 = buildTestTrace(ownerDW2);
+        processor.processProgramTrace(trace2, hash2);
+
+        LinkedList<DataWord> ownerDWList = new LinkedList<>();
+        ownerDWList.add(ownerDW);
+        ownerDWList.add(ownerDW2);
 
-        Assert.assertNotNull(jnode);
-        Assert.assertEquals(ByteUtil.toHexString(ownerDW.getLast20Bytes()), jnode.get("contractAddress").asText());
+        JsonNode jsonNode = processor.getProgramTraceAsJsonNode(hash);
 
-        String jsonText = jnode.toString();
+        Assert.assertNotNull(jsonNode);
+        Assert.assertEquals(ByteUtil.toHexString(ownerDW.getLast20Bytes()), jsonNode.get("contractAddress").asText());
+
+        String jsonText = jsonNode.toString();
         Assert.assertNotNull(jsonText);
         Assert.assertTrue(jsonText.contains("\"contractAddress\""));
         Assert.assertTrue(jsonText.contains("\"initStorage\""));
         Assert.assertTrue(jsonText.contains("\"currentStorage\""));
         Assert.assertTrue(jsonText.contains("\"structLogs\""));
         Assert.assertTrue(jsonText.contains("\"result\""));
         Assert.assertTrue(jsonText.contains("\"storageSize\""));
+
+        jsonNode = processor.getProgramTracesAsJsonNode(Arrays.asList(hash, hash2));
+
+        Assert.assertNotNull(jsonNode);
+        Assert.assertTrue(jsonNode.isArray());
+        ArrayNode arrNode = (ArrayNode) jsonNode;
+        Assert.assertEquals(2, arrNode.size());
+        arrNode.forEach(jn -> {
+            DataWord owner = ownerDWList.poll();
+            Assert.assertNotNull(owner);
+            Assert.assertNotNull(jn);
+            Assert.assertEquals(ByteUtil.toHexString(owner.getLast20Bytes()), jn.get("contractAddress").asText());
+
+            String jsonStr = jn.toString();
+            Assert.assertNotNull(jsonStr);
+            Assert.assertTrue(jsonStr.contains("\"contractAddress\""));
+            Assert.assertTrue(jsonStr.contains("\"initStorage\""));
+            Assert.assertTrue(jsonStr.contains("\"currentStorage\""));
+            Assert.assertTrue(jsonStr.contains("\"structLogs\""));
+            Assert.assertTrue(jsonStr.contains("\"result\""));
+            Assert.assertTrue(jsonStr.contains("\"storageSize\""));
```

Comment:
Maybe something like this to make the test more extendable to new fields? (Since `List.of` is still a random sun dependency and it's not 'canon' until Java 9 I think, maybe ignore this)
```java
            List<String> requiredFields = List.of("contractAddress", "initStorage", "currentStorage", "structLogs", "result", "storageSize");
            boolean requiredFieldsArePresent = requiredFields.stream()
                .allMatch(field -> jn.findValue(field) != null);
            Assert.assertTrue(requiredFieldsArePresent);
```

Reply:
done